drivers/hmis/lib/hmis_util/json_forms.rb

```diff
@@ -173,13 +177,36 @@ def apply_patches(tree, patches)
       [result, applied_patches]
     end
 
+    # Similar to apply_item_patches, but applies patches to the overall form
+    # instead of to an individual item. This was added to support appending an item
+    # to the Service form.
+    def apply_form_patches(definition, patches, identifier:)
+      result = definition.deep_dup
+      applied_patches = []
+      patches.filter { |patch| patch['form_identifier'] == identifier && !patch.key?('link_id')}.each do |patch|
```

Comment:
seems like the caller already does this filtering. Maybe raise if extra safety is needed
```suggestion
      patches.filter { |patch| patch['form_identifier'] == identifier }.each do |patch|
        raise if patch.key?('link_id') # should not reach this
```